pkg/scheduler/framework/interface.go

```diff
@@ -292,7 +291,7 @@ func (p PluginToStatus) Merge() *Status {
 
 	finalStatus := NewStatus(Success)
 	for _, s := range p {
-		if s.Code() == Error {
+		if s.Code() == Error || s.err != nil {
 			finalStatus.err = s.AsError()
 		}
```

Comment:
Maybe this is enough?
```suggestion
		finalStatus.err = s.err
```

Reply:
x:    AsStatus(errors.New("1")),
y:    AsStatus(errors.New("2"))

currently, if we merge x and y with `Merge()`, the final status will be look like:
```
Status{
	code:         Error,
	reasons:      nil,
	err:          x.err,
	failedPlugin: "",
}
```
We'll loose some error msg here. So I think when merging two error status, we should append error msgs to reasons too.

Reply:
I've thought about this before, but I think we should not have two error status in reality. Once we met an error in scheduling cycle, we'll return directly.